### ISSUE TITLE
Permission errors for precompilation on Windows

### ISSUE THREAD

**Issue by fredrikekre** [JuliaLang/julia#38411]
Several users report file permission errors when precompiling on Julia 1.5.3 which did not happen on 1.5.2, see https://discourse.julialang.org/t/precompilation-error-with-julia-1-5-3-windows-10-64-bit/50028. Since there was a libuv upgrade (rebuild?) between those to releases I guess that might be related.

**Comment by bkamins** [comment]
I confirm this problem. Currently under Win10 you need to run Julia with administrator privileges to be able to precompile packages (this happens if you run it on top of files previously generated by Julia 1.5.2, so maybe in a clean environment this does not happen - I have not checked).

**Comment by KristofferC** [comment]
Well, that is quite unfortunate :/. https://github.com/JuliaLang/julia/commit/0305b7c1f84ea68d654d2f2f86b545e6a85470ef did something with permissions but I don't at glance see how that could cause this.

**Comment by bkamins** [comment]
What is the status of the issue? I think it requires a quick patch (but unfortunately I am probably not the best person to try fixing it).
I had a teaching session today and the problem also appeared.

Just to expand on the issue: this error shows also when you update packages under Windows on 1.5.3.

**Comment by fredrikekre** [comment]
So the problem seems to be that precompilation files created with 1.5.2 have "wrong" permissions then? Does the 1.5.2 installer have elevated permissions somehow or how can that binary create files that 1.5.3 is not allowed to change?

**Comment by KristofferC** [comment]
Does clearing `.julia/compiled` actually fix it? It seems people reported errors even after that, and only using 1.5.3?

**Comment by bkamins** [comment]
No it does not. I have removed this folder and then did:
```
using Tables
```
what happened is that:
1. DataValuesInterfaces.jl compiled correctly (it is a dependency) - a folder for it got created
2. Tables.jl errors (see below) - a folder for it got created but is empty (it has the same permissions as the folder for DataValuesInterfaces.jl though)

```
julia> using Tables
[ Info: Precompiling Tables [bd369af6-aec1-5ad0-b16a-f7cc5008161c]
ERROR: LoadError: IOError: unlink: permission denied (EACCES)
Stacktrace:
 [1] uv_error at .\libuv.jl:97 [inlined]
 [2] unlink(::String) at .\file.jl:918
 [3] rm(::String; force::Bool, recursive::Bool) at .\file.jl:268
 [4] compilecache(::Base.PkgId, ::String) at .\loading.jl:1300
 [5] _require(::Base.PkgId) at .\loading.jl:1030
 [6] require(::Base.PkgId) at .\loading.jl:928
 [7] require(::Module, ::Symbol) at .\loading.jl:923
 [8] include(::Function, ::Module, ::String) at .\Base.jl:380
 [9] include(::Module, ::String) at .\Base.jl:368
 [10] top-level scope at none:2
 [11] eval at .\boot.jl:331 [inlined]
 [12] eval(::Expr) at .\client.jl:467
 [13] top-level scope at .\none:3
in expression starting at D:\AppData\.julia\packages\Tables\GBP4H\src\Tables.jl:3
caused by [exception 1]
IOError: unlink: permission denied (EACCES)
Stacktrace:
 [1] uv_error at .\libuv.jl:97 [inlined]
 [2] unlink(::String) at .\file.jl:918
 [3] rm(::String; force::Bool, recursive::Bool) at .\file.jl:268
 [4] rename(::String, ::String; force::Bool) at .\file.jl:928
 [5] compilecache(::Base.PkgId, ::String) at .\loading.jl:1296
 [6] _require(::Base.PkgId) at .\loading.jl:1030
 [7] require(::Base.PkgId) at .\loading.jl:928
 [8] require(::Module, ::Symbol) at .\loading.jl:923
 [9] include(::Function, ::Module, ::String) at .\Base.jl:380
 [10] include(::Module, ::String) at .\Base.jl:368
 [11] top-level scope at none:2
 [12] eval at .\boot.jl:331 [inlined]
 [13] eval(::Expr) at .\client.jl:467
 [14] top-level scope at .\none:3
┌ Warning: temp cleanup
│   exception =
│    IOError: unlink: permission denied (EACCES)
│    Stacktrace:
│     [1] uv_error at .\libuv.jl:97 [inlined]
│     [2] unlink(::String) at .\file.jl:918
│     [3] rm(::String; force::Bool, recursive::Bool) at .\file.jl:268
│     [4] temp_cleanup_purge(::Bool) at .\file.jl:514
│     [5] temp_cleanup_purge() at .\file.jl:507
│     [6] _atexit() at .\initdefs.jl:316
│     [7] exit at .\initdefs.jl:28 [inlined]
│     [8] _start() at .\client.jl:509
└ @ Base.Filesystem file.jl:518
ERROR: Failed to precompile Tables [bd369af6-aec1-5ad0-b16a-f7cc5008161c] to D:\AppData\.julia\compiled\v1.5\Tables\Z804B_MpOTO.ji.
Stacktrace:
 [1] error(::String) at .\error.jl:33
 [2] compilecache(::Base.PkgId, ::String) at .\loading.jl:1305
 [3] _require(::Base.PkgId) at .\loading.jl:1030
 [4] require(::Base.PkgId) at .\loading.jl:928
 [5] require(::Module, ::Symbol) at .\loading.jl:923
```

**Comment by KristofferC** [comment]
https://github.com/JuliaLang/julia/issues/38433 seems related.

Is it this `chmod` that messes things up https://github.com/JuliaLang/julia/blob/599ecd82101ff7df1bd68db2c9d56410358907a9/base/loading.jl#L1293?

**Comment by fredrikekre** [comment]
Aren't we `chmod`ing now when cleaning up tmpdirs?

**Comment by KristofferC** [comment]
Hm, yes, so maybe it is chmod in both cases...?

**Comment by bkamins** [comment]
Any progress on this. For a reference this is what I get when trying to do `using Parsers`:
```
Path   : .julia\compiled\v1.5\Parsers\833b9_MpOTO.ji
Access : BUILTIN\Users Allow  Modify, Synchronize
         BUILTIN\Administrators Allow  FullControl
         NT AUTHORITY\SYSTEM Allow  FullControl
         NT AUTHORITY\Authenticated Users Allow  Modify, Synchronize

Path   : .julia\compiled\v1.5\Parsers\jl_5C04.tmp
Access : Everyone Allow  DeleteSubdirectoriesAndFiles, Write, ReadAndExecute, Synchronize
         NT AUTHORITY\Authenticated Users Allow  DeleteSubdirectoriesAndFiles, Write, ReadAndExecute, Synchronize
         NT AUTHORITY\SYSTEM Allow  FullControl
         BUILTIN\Administrators Allow  FullControl
         BUILTIN\Users Allow  DeleteSubdirectoriesAndFiles, Write, ReadAndExecute, Synchronize
         HOST\username Allow  DeleteSubdirectoriesAndFiles, Write, ReadAndExecute, Synchronize
```
as you can see the problem is with `jl_5C04.tmp` file. The `833b9_MpOTO.ji` file has correct permissions. For the temp file the problem is that `Modify` right is not given to the process as you can see, which disallows the Julia process to delete the file (unless run as administrator).

**Comment by KristofferC** [comment]
I don't think we understand why this happens and attempts to reproduce it by other people have failed. I think what is needed is the exact point where the divergence in behavior from the earlier version happens.

**Comment by pemryan** [comment]
encountered this permission error during the compiling process in cygwin-64. The julia.exe excuatable has been generated successfully and then failed in the julia precompilation.

```
make[1]: Entering directory '/cygdrive/c/Users/pem/Downloads/julia-1.5.3/julia-win64'
Warning: git information unavailable; versioning information limited
    JULIA julia-win64/usr/lib/julia/sys-o.a
ERROR: LoadError: IOError: mkdir: permission denied (EACCES)
Stacktrace:
 [1] uv_error at .\libuv.jl:97 [inlined]
 [2] mkdir(::String; mode::UInt16) at .\file.jl:177
 [3] mkpath(::String; mode::UInt16) at .\file.jl:227
 [4] mkpath(::String; mode::UInt16) at .\file.jl:225
 [5] mkpath at .\file.jl:222 [inlined]
 [6] (::Main.anonymous.var"#1#5")(::String) at C:\Users\pem\Downloads\julia-1.5.3\contrib\generate_precompile.jl:76
 [7] mktempdir(::Main.anonymous.var"#1#5", ::String; prefix::String) at .\file.jl:709
 [8] mktempdir at .\file.jl:707 [inlined] (repeats 2 times)
 [9] generate_precompile_statements() at C:\Users\pem\Downloads\julia-1.5.3\contrib\generate_precompile.jl:72
 [10] top-level scope at C:\Users\pem\Downloads\julia-1.5.3\contrib\generate_precompile.jl:202
in expression starting at C:\Users\pem\Downloads\julia-1.5.3\contrib\generate_precompile.jl:3
*** This error is usually fixed by running `make clean`. If the error persists, try `make cleanall`. ***
make[1]: *** [/cygdrive/c/Users/pem/Downloads/julia-1.5.3/sysimage.mk:86: /cygdrive/c/Users/pem/Downloads/julia-1.5.3/julia-win64/usr/lib/julia/sys-o.a] Error 1
make[1]: Leaving directory '/cygdrive/c/Users/pem/Downloads/julia-1.5.3/julia-win64'
make: *** [/cygdrive/c/Users/pem/Downloads/julia-1.5.3/Makefile:87: julia-sysimg-release] Error 2
make: Leaving directory '/cygdrive/c/Users/pem/Downloads/julia-1.5.3/julia-win64'

```

**Comment by pemryan** [comment]
chmod doesn't work. make clean and re-compile will fail into the same error.

**Comment by KristofferC** [comment]
If you revert https://github.com/JuliaLang/julia/pull/38159 does it still happen?

**Comment by pemryan** [comment]
> 
> 
> If you revert [a3430e7](https://github.com/JuliaLang/julia/commit/a3430e7ec1d6c2cc883b7e5079c1a0e096f2eb66) does it still happen?

It doesn't.

For my precompilation  error case, it seems to be a conflict between cygwin's persmission system and windows ACL. 
by changing cygwin's /etc/fstab and adding 'noacl' flag:
`none /cygdrive cygdrive binary,noacl,posix=0,user 0 0`
and changing the target folder's default permission setttings (grant "User", "ALL APPLICATION PACKAGES" and ''CREATOR OWNER" full access).  All is going smoothly now.

**Comment by KristofferC** [comment]
> For offical julia 1.5.3 binaries, the precompilation error when I tried to import/using a package was also solved by changing the DEPOT folder's default permission setttings.

If you delete it completely, does the new one we create have the correct permissions?

**Comment by bkamins** [comment]
> does the new one we create have the correct permissions?

In my case no. I am not sure if I have explicitly given this information above, but deleting everything and re-instantiating everything from scratch does not solve the problem.

**Comment by thofma** [comment]
Will this be fixed for 1.6?

**Comment by KristofferC** [comment]
If someone figures out how to do it. It is still unclear, why this only happens on some Windows machines and what is the change that caused it.

**Comment by yilmazdurmaz** [comment]
I got similar error for Plots package and filed an issue thinking the problem belonged to it. Then I got more errors for more packages so I found this issue here. Now I know it is part of  1.5.3.

**Comment by musm** [comment]
@yilmazdurmaz  if you unset the JULIA_DEPOT_PATH does it still error?

**Comment by yilmazdurmaz** [comment]
@musm  I have tested in these locations set as `JULIA_DEPOT_PATH`:
```
T:\Tools\julia-1.5.3\.julia
C:\.julia\
C:\Users\durmaz\.julia
C:\Users\durmaz\Download\.julia
```
If it is under user's folder, it has no problem on precompilations. outside, it gives exactly same errors for different packages (other than few lines of package name-path) as in [bkamin's comment](https://github.com/JuliaLang/julia/issues/38411#issuecomment-726851534) above

after each test, I tried to remove these folders and I found out that even though Julia is started in a non-admin console, for non-user paths, many files/folders are managed to end up admin-level files, in which admin rights required to remove them. this had to not be happened without me giving any admin rights to that console.

as stated in previous comments, 1.5.2 does not have this problem.

**Comment by musm** [comment]
Many thanks I can reproduce the issue now.

Indeed, the issue is that if `JULIA_DEPOT_PATH` is set to a location outside the user's folder, the permissions on `.julia` are set incorrectly (user permissions are not enabled)

Repro: 
1. First set JULIA_DEPOTH_PATH to, e.g., C:\.julia (or any other folder outside the user's folder)
2. Start Julia and add, e.g. `pkg> Plots`

**Comment by yilmazdurmaz** [comment]
@musm error won't show itself until try to use by "`Using PackageName`"

**Comment by musm** [comment]
Ok so basically the issue is that unless you manually create  the folder pointed to `JULIA_DEPOT_PATH`  with correct permissions, the folder that is created in a non-user directory (e.g. `C:\`)  inherents  the permissions of the parent directory. 

I believe we would need to change the security descriptor the created folder in `mkdir`, but unfortunately  libuv doesn't implement the `mode` option for Windows yet, so that is a no-op.

**Comment by StefanKarpinski** [comment]
Still worth keeping an issue open though so that we can track the issue, even if we can't currently fix it.

**Comment by musm** [comment]
That was a mistake, thanks for re-opening

**Comment by KristofferC** [comment]
So what made this happen in 1.5.3?

**Comment by yilmazdurmaz** [comment]
TL:DR: please read the second part below to narrow down the issue

Let me recap the issues:
- v1.5.2 has no problem
- v1.5.3 does something different at some point
- issue can be recrated by 
  - `]add PackageName`
  - `using PackageName`  <- errors shown after this
- issue happens in non-user directories but also these directories are created by users, not by an admin. and Julia is started in user mode, not in admin mode.
- trying to delete ".julia" folder, most if not all packages' **files** under "packages" folder (but not the folders themselves) require admin rights to be deleted.

This last item in the list is new. I tested files and folders under the DEPOT folder. the problem is not about folders, but files of downloaded packages.
- only the **files** are affected under the `packages` subfolder. none of the other files/folders are affected.
- These files have only "read" property set for anyone other than admin.
- the problem seems to occur only while downloading packages, and to the packages' downloaded files only. none of subfolders are affected.
- "build.log" of the package added after the downloading is not affected.

this should narrow down your search for the root of the problem.

**Comment by musm** [comment]
https://github.com/JuliaLang/julia/pull/38942 should fix this

**Comment by musm** [comment]
Yeah 1.5.3 apparently had a libuv bump which changed the underlying `mkdir` method in libuv

**Comment by musm** [comment]
It turns out that the offending commits are 
https://github.com/JuliaLang/libuv/commit/794bfdfd22bf50f9fa9cc0a83
https://github.com/JuliaLang/libuv/commit/1dcf32479840be96210d0807037351626cb82e2f

**Comment by musm** [comment]
@staticfloat any chance when you rebased your commits on the 1.39 fork could've accidentally introduced a bug?

**Comment by staticfloat** [comment]
Hmmm, yes since those commits are changing what `chmod()` does on Windows it's entirely possible that the issue is my fault.  In essence, what the libuv commits do is change `chmod()` to apply ACLs on Windows to mimic what we are passing `chmod()`. (Note that `stat()` doesn't report these back properly, and it's not clear if it ever will.  If you want to see the permissions, you should either inspect them inside of Explorer or use `icacls`)

I would be curious to see the `icacls` output of a file you get a permission error on.

**Comment by yilmazdurmaz** [comment]
solved and closed?!?
or closed by accident again?

**Comment by yilmazdurmaz** [comment]
@staticfloat 

I tried to use ``SQLite`` package in clean environment made from zipped Julia distribution. From error messages, ``Sqlite`` tries to import ``WeakRefStrings`` which in turn tries to import ``DataAPI``. The ``compiled`` folder has nothing else in it after the error so I have the "`icacls`" outputs for following folder tree. I also included comparison for 1.5.2 where all folders have same signature but file signatures are changed (user cannot delete files inside packages folder in 1.5.3 while everything is deletable in 1.5.2).

````
T:\TOOLS\JULIA-1.5.3\.JULIA\
└───COMPILED
│   ├───v1.5
│       ├───DataAPI
│       │   │   3a8mN_3hQs8.ji  (created during compilation, user can delete)
│       │   │   jl_330D.tmp (created during compilation, user cannot delete, can't say specific to 1.5.3)
│       ├───SQLite
│       └───WeakRefStrings
└───PACKAGES
    ├───SQLITE
        ├───LjMUn
            │   README.md (created while adding package, user cannot delete in 1.5.3)
    ├───DATAAPI
        ├───LjMUn
            │   README.md (created while adding package, user cannot delete in 1.5.3)
````



````
t:\Tools\julia-1.5.3\.julia\compiled\v1.5\DataAPI
  BUILTIN\Administrators:(I)(F)
  BUILTIN\Administrators:(I)(OI)(CI)(IO)(F)
  NT AUTHORITY\SYSTEM:(I)(F)
  NT AUTHORITY\SYSTEM:(I)(OI)(CI)(IO)(F)
  NT AUTHORITY\Authenticated Users:(I)(M)
  NT AUTHORITY\Authenticated Users:(I)(OI)(CI)(IO)(M)
  BUILTIN\Users:(I)(RX)
  BUILTIN\Users:(I)(OI)(CI)(IO)(GR,GE)

t:\Tools\julia-1.5.3\.julia\compiled\v1.5\DataAPI\3a8mN_3hQs8.ji
  BUILTIN\Administrators:(I)(F)
  NT AUTHORITY\SYSTEM:(I)(F)
  NT AUTHORITY\Authenticated Users:(I)(M)
  BUILTIN\Users:(I)(RX)

t:\Tools\julia-1.5.3\.julia\compiled\v1.5\DataAPI\jl_330D.tmp
  SOULS-DX\durmaz:(RX,W,DC)
  Everyone:(RX,W,DC)
  NT AUTHORITY\Authenticated Users:(RX,W,DC)
  BUILTIN\Users:(RX,W,DC)
  BUILTIN\Administrators:(F)
  NT AUTHORITY\SYSTEM:(F)


t:\Tools\julia-1.5.3\.julia\packages\SQLite\LjMUn
  BUILTIN\Administrators:(I)(F)
  BUILTIN\Administrators:(I)(OI)(CI)(IO)(F)
  NT AUTHORITY\SYSTEM:(I)(F)
  NT AUTHORITY\SYSTEM:(I)(OI)(CI)(IO)(F)
  NT AUTHORITY\Authenticated Users:(I)(M)
  NT AUTHORITY\Authenticated Users:(I)(OI)(CI)(IO)(M)
  BUILTIN\Users:(I)(RX)
  BUILTIN\Users:(I)(OI)(CI)(IO)(GR,GE)

t:\Tools\julia-1.5.3\.julia\packages\SQLite\LjMUn\README.md
  SOULS-DX\durmaz:(R,WA)
  Everyone:(R,WA)
  NT AUTHORITY\Authenticated Users:(R,WA)
  BUILTIN\Users:(R,WA)
  BUILTIN\Administrators:(F)
  NT AUTHORITY\SYSTEM:(F)

t:\Tools\julia-1.5.3\.julia\packages\DataAPI\GOXrJ\README.md
  SOULS-DX\durmaz:(R,WA)
  Everyone:(R,WA)
  NT AUTHORITY\Authenticated Users:(R,WA)
  BUILTIN\Users:(R,WA)
  BUILTIN\Administrators:(F)
  NT AUTHORITY\SYSTEM:(F)
````

And here are the same files for 1.5.2 (except the tmp file)

````
t:\Tools\julia-1.5.2\.julia\compiled\v1.5\DataAPI
  BUILTIN\Administrators:(I)(F)
  BUILTIN\Administrators:(I)(OI)(CI)(IO)(F)
  NT AUTHORITY\SYSTEM:(I)(F)
  NT AUTHORITY\SYSTEM:(I)(OI)(CI)(IO)(F)
  NT AUTHORITY\Authenticated Users:(I)(M)
  NT AUTHORITY\Authenticated Users:(I)(OI)(CI)(IO)(M)
  BUILTIN\Users:(I)(RX)
  BUILTIN\Users:(I)(OI)(CI)(IO)(GR,GE)

t:\Tools\julia-1.5.2\.julia\compiled\v1.5\DataAPI\3a8mN_qfkbB.ji
  BUILTIN\Administrators:(I)(F)
  NT AUTHORITY\SYSTEM:(I)(F)
  NT AUTHORITY\Authenticated Users:(I)(M)
  BUILTIN\Users:(I)(RX)

t:\Tools\julia-1.5.2\.julia\packages\SQLite\LjMUn
  BUILTIN\Administrators:(I)(F)
  BUILTIN\Administrators:(I)(OI)(CI)(IO)(F)
  NT AUTHORITY\SYSTEM:(I)(F)
  NT AUTHORITY\SYSTEM:(I)(OI)(CI)(IO)(F)
  NT AUTHORITY\Authenticated Users:(I)(M)
  NT AUTHORITY\Authenticated Users:(I)(OI)(CI)(IO)(M)
  BUILTIN\Users:(I)(RX)
  BUILTIN\Users:(I)(OI)(CI)(IO)(GR,GE)

t:\Tools\julia-1.5.2\.julia\packages\SQLite\LjMUn\README.md
  BUILTIN\Administrators:(I)(F)
  NT AUTHORITY\SYSTEM:(I)(F)
  NT AUTHORITY\Authenticated Users:(I)(M)
  BUILTIN\Users:(I)(RX)

t:\Tools\julia-1.5.2\.julia\packages\DataAPI\GOXrJ\README.md
  BUILTIN\Administrators:(I)(F)
  NT AUTHORITY\SYSTEM:(I)(F)
  NT AUTHORITY\Authenticated Users:(I)(M)
  BUILTIN\Users:(I)(RX)
````

**Comment by KristofferC** [comment]
Why reopen?

**Comment by musm** [comment]
I'm not confident the correct solution is to uniformly `chmod` the folder created as in https://github.com/JuliaLang/julia/pull/38942 , which I think we will have to revert.  One issue with that PR is folder's created in e.g. 
```
mkdir("C:\\test")
```
Can no longer be `rm("C:\\test")` due to permission errors. With that patch.

**Comment by KristofferC** [comment]
Well, if we have to revert it, then let's revert it? I didn't get the impression from the PR description that you didn't actually want it to get merged.

**Comment by musm** [comment]
No fault of your own, I should have made it more clear.  I wasn't sure on the exact crux of the issue until today and it became clear that the PR wasn't the exact triggering point.

**Comment by yilmazdurmaz** [comment]
I went ahead and tried to import `SQLite` twice, and find a workaround solution.

- first try is to delete every `*.tmp` file created during compilation. about 10 tries,  each time error is on another file, but eventually reached to import SQLite.
- without any removal, I just used `using SQLite` about 10 times repeatedly. each error was on a different file but eventually imported SQLite.

so workaround basically comes down to this:
- precompilation gives error because removal of `*.tmp` file fails but it also succeeds on precompiling the package so that next time it needs same package, it wont try recompilation and wont fail again for same package.
- repeating same `using` statement will eventually succeed to precompiling all inner imports and then the package we want to import.

there is one thing I need to be clear: although the error can be fixed in precompilation method, I mentioned all the time that all files in packages's folder has the same permissions error. this is because the same submethod is possibly used in `Pkg` and both should be fixed together.

**Comment by bkamins** [comment]
Now precompilation works for me under Windows in terminal without a problem. However I today hit the following issue:
```
(@v1.6) pkg> add PooledArrays
    Updating registry at `D:\.julia\registries\General`
    Updating git-repo `https://github.com/JuliaRegistries/General.git`
   Resolving package versions...
   Installed PooledArrays ─ v1.2.1
    Updating `D:\.julia\environments\v1.6\Project.toml`
  [2dfb63ee] + PooledArrays v1.2.1
    Updating `D:\.julia\environments\v1.6\Manifest.toml`
  [2dfb63ee] ↑ PooledArrays v0.5.3 ⇒ v1.2.1
Precompiling project...
  Progress [>                                        ]  0/3
  ◓ PooledArrays
┌ Error: Pkg.precompile error
│   exception =
│    IOError: chmod("D:\\.julia\\compiled\\v1.6\\PooledArrays\\vi11X_2OVCs.ji", 0o777): unknown error (UNKNOWN)
│    Stacktrace:
│      [1] uv_error
│        @ .\libuv.jl:97 [inlined]
│      [2] chmod(path::String, mode::UInt16; recursive::Bool)
│        @ Base.Filesystem .\file.jl:1049
│      [3] chmod
│        @ .\file.jl:1048 [inlined]
│      [4] rm(path::String; force::Bool, recursive::Bool)
│        @ Base.Filesystem .\file.jl:269
│      [5] checkfor_mv_cp_cptree(src::String, dst::String, txt::String; force::Bool)
│        @ Base.Filesystem .\file.jl:310
│      [6] cp(src::String, dst::String; force::Bool, follow_symlinks::Bool)
│        @ Base.Filesystem .\file.jl:349
│      [7] rename(src::String, dst::String; force::Bool)
│        @ Base.Filesystem .\file.jl:943
│      [8] compilecache(pkg::Base.PkgId, path::String, internal_stderr::IOBuffer, internal_stdout::Base.DevNull)
│        @ Base .\loading.jl:1347
│      [9] (::Pkg.API.var"#212#237"{IOBuffer, String, Base.PkgId})()
│        @ Pkg.API C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\API.jl:1132
│     [10] with_logstate(f::Function, logstate::Any)
│        @ Base.CoreLogging .\logging.jl:491
│     [11] with_logger
│        @ .\logging.jl:603 [inlined]
│     [12] macro expansion
│        @ C:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.6\Pkg\src\API.jl:1131 [inlined]
│     [13] (::Pkg.API.var"#209#234"{Bool, Vector{Task}, Pkg.API.var"#handle_interrupt#227"{Base.Event, ReentrantLock, Base.TTY}, Pkg.API.var"#color_string#225", Base.Event, Base.Event, ReentrantLock, Vector{Base.PkgId}, Vector{Base.PkgId}, Dict{Base.PkgId, String}, Vector{Base.PkgId}, Vector{Base.PkgId}, Dict{Base.PkgId, Bool}, Dict{Base.PkgId, Base.Event}, Dict{Base.PkgId, Bool}, Dict{Base.UUID, Pkg.Types.PackageEntry}, Vector{Base.PkgId}, Bool, Base.TTY, Base.Semaphore, String, Vector{String}, Vector{Base.PkgId}, Base.PkgId})()
  Progress [>                                        ]  0/1
  ◒ PooledArrays
```
on:
```
julia> versioninfo()
Julia Version 1.6.0-rc1
Commit a58bdd9010 (2021-02-06 15:49 UTC)
Platform Info:
  OS: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-11.0.1 (ORCJIT, skylake)
```
when run in VSCode terminal.

(I precompiled the package in the normal terminal after this without a problem)
